## Final Project Submission

Please fill out:
* Student name: Cindy Minyade
* Student pace: part time 
* Scheduled project review date/time: 
* Instructor name: Antony Muiko
* Blog post URL:


## AIRCRAFT RISK ASSESSMENT AND RECOMMENDATION FOR COMMERCIAL BUSINESS EXPANSION

## 1. Project Goal
My project aims to determine which aircrafts are the lowest risk for the company to start this new business endeavor

## 2. Data Source and Data Exploration
This data comes from Kaggle website. It is from the National Transportation Safety Board that includes aviation accident data from 1962 to 2023 about civil aviation accidents and selected incidents in the United States and international waters. 

## 3. Data Exploration

# Load the python libraries 
To begin analyzing the dataset, I will import the Pandas library to load the data and clean it.I will also import other libraries that may become useful during data cleaning and visualization. 


In [76]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline


# Loading the data
In the cell below, loading `Aviation_Data.csv` as `aviation_df`
 

In [77]:
# loading data
aviation_df = pd.read_csv("data/Aviation_Data.csv", index_col=0)

# Display the first 5 rows
aviation_df.head(5)


/var/folders/14/11n8463108j28v64ydsd08680000gn/T/ipykernel_46154/3921986474.py:2: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  aviation_df = pd.read_csv("data/Aviation_Data.csv", index_col=0)


Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001218X45444           Accident      SEA87LA080  1948-10-24   
20001218X45447           Accident      LAX94LA336  1962-07-19   
20061025X01555           Accident      NYC07LA005  1974-08-30   
20001218X45448           Accident      LAX96LA321  1977-06-19   
20041105X01764           Accident      CHI79FA064  1979-08-02   

                       Location        Country   Latitude  Longitude  \
Event.Id                                                               
20001218X45444  MOOSE CREEK, ID  United States        NaN        NaN   
20001218X45447   BRIDGEPORT, CA  United States        NaN        NaN   
20061025X01555    Saltville, VA  United States  36.922223 -81.878056   
20001218X45448       EUREKA, CA  United States        NaN        NaN   
20041105X01764       Canton, OH  United States        NaN        NaN   

               Airport.Code Airport.Name Injury.Severity  ...  \
Event.Id                                                  ...   
20001218X45444          NaN          NaN        Fatal(2)  ...   
20001218X45447          NaN          NaN        Fatal(4)  ...   
20061025X01555          NaN          NaN        Fatal(3)  ...   
20001218X45448          NaN          NaN        Fatal(2)  ...   
20041105X01764          NaN          NaN        Fatal(1)  ...   

               Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
Event.Id                                                            
20001218X45444          Personal         NaN                  2.0   
20001218X45447          Personal         NaN                  4.0   
20061025X01555          Personal         NaN                  3.0   
20001218X45448          Personal         NaN                  2.0   
20041105X01764          Personal         NaN                  1.0   

               Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
Event.Id                                                                     
20001218X45444                    0.0                  0.0             0.0   
20001218X45447                    0.0                  0.0             0.0   
20061025X01555                    NaN                  NaN             NaN   
20001218X45448                    0.0                  0.0             0.0   
20041105X01764                    2.0                  NaN             0.0   

                Weather.Condition Broad.phase.of.flight   Report.Status  \
Event.Id                                                                  
20001218X45444                UNK                Cruise  Probable Cause   
20001218X45447                UNK               Unknown  Probable Cause   
20061025X01555                IMC                Cruise  Probable Cause   
20001218X45448                IMC                Cruise  Probable Cause   
20041105X01764                VMC              Approach  Probable Cause   

               Publication.Date  
Event.Id                         
20001218X45444              NaN  
20001218X45447       19-09-1996  
20061025X01555       26-02-2007  
20001218X45448       12-09-2000  
20041105X01764       16-04-1980  

[5 rows x 30 columns]

Now to farmiliarize with the data:  
- Understand the dimentions  
- Type of data it contains  
- If there's missing values  

In [78]:
#type
type(aviation_df)

pandas.core.frame.DataFrame

In [79]:
aviation_df.shape

(90348, 30)

In [80]:
#dimentions
aviation_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 90348 entries, 20001218X45444 to 20221230106513
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Investigation.Type      90348 non-null  object 
 1   Accident.Number         88889 non-null  object 
 2   Event.Date              88889 non-null  object 
 3   Location                88837 non-null  object 
 4   Country                 88663 non-null  object 
 5   Latitude                34382 non-null  object 
 6   Longitude               34373 non-null  object 
 7   Airport.Code            50132 non-null  object 
 8   Airport.Name            52704 non-null  object 
 9   Injury.Severity         87889 non-null  object 
 10  Aircraft.damage         85695 non-null  object 
 11  Aircraft.Category       32287 non-null  object 
 12  Registration.Number     87507 non-null  object 
 13  Make                    88826 non-null  object 
 14  Model                

# Interpretation of the data information
- the data set has 31 columns, 90348 rows
- data has missing values in most columns
- data has 5 columns with float and 25 with string data types

# Performing Data Cleaning 
_What are the aircrafts with the lowest risk?_

Cleaning and Filtering data


In [81]:
#determine the column names

column_names = aviation_df.columns
column_names

Index(['Investigation.Type', 'Accident.Number', 'Event.Date', 'Location',
       'Country', 'Latitude', 'Longitude', 'Airport.Code', 'Airport.Name',
       'Injury.Severity', 'Aircraft.damage', 'Aircraft.Category',
       'Registration.Number', 'Make', 'Model', 'Amateur.Built',
       'Number.of.Engines', 'Engine.Type', 'FAR.Description', 'Schedule',
       'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [82]:
#selecting the relevant columns that we will use for our data anylsis
selected_columns = [
    'Make', 'Model', 'Aircraft.damage', 'Aircraft.Category', 'Total.Fatal.Injuries', 'Broad.phase.of.flight', 'Total.Serious.Injuries', 'Weather.Condition'
]

print(selected_columns)

['Make', 'Model', 'Aircraft.damage', 'Aircraft.Category', 'Total.Fatal.Injuries', 'Broad.phase.of.flight', 'Total.Serious.Injuries', 'Weather.Condition']


In [83]:
#check for missing data in the selected columns
missing_values = aviation_df[selected_columns].isna().sum()
print(missing_values)


Make                       1522
Model                      1551
Aircraft.damage            4653
Aircraft.Category         58061
Total.Fatal.Injuries      12860
Broad.phase.of.flight     28624
Total.Serious.Injuries    13969
Weather.Condition          5951
dtype: int64


In [84]:
# drop the rows with the missing values from the selected columns
clean_aviation_df = aviation_df[selected_columns].dropna()


## Data Analysis

In [95]:
#Grouping the data by aircraft make and model to determine the crafts incidents
damage_summary = (
    clean_aviation_df.groupby(["Make", "Model", "Aircraft.Category", "Aircraft.damage"]) #group data by make, model, and aircraft damage
).size().reset_index(name="Incident.Count")   # to find how many incidents ocurred in each groupedby dataset

damage_summary


Make          Model Aircraft.Category Aircraft.damage  \
0                    Adams            A55           Balloon     Substantial   
1           Aero Commander            100          Airplane       Destroyed   
2           Aero Commander            100          Airplane     Substantial   
3           Aero Commander            112          Airplane     Substantial   
4           Aero Commander          112TC          Airplane     Substantial   
...                    ...            ...               ...             ...   
1562             Weatherly           201C          Airplane     Substantial   
1563                  Wing            D-1          Airplane       Destroyed   
1564         Wootton, John    Aero Canard          Airplane     Substantial   
1565  Wrb Associates, Llc.  Zodiac 601 XL          Airplane       Destroyed   
1566              Yakovlev         Yak 52          Airplane     Substantial   

      Incident.Count  
0                  1  
1                  1  
2                  3  
3                  1  
4                  1  
...              ...  
1562               4  
1563               1  
1564               1  
1565               1  
1566               1  

[1567 rows x 5 columns]

In [96]:
# aggregate the data of injuries occured, to have the total number of serious and fatal injuries
injury_summary = (
    clean_aviation_df.groupby(["Make", "Model", "Aircraft.Category"]) #group data by aircraft made and model
    .agg({
        "Total.Fatal.Injuries":"sum", "Total.Serious.Injuries":"sum"   # sum the fatal and serious injuries
    })
    .reset_index()  #reset the index
)


In [97]:
injury_summary

Make          Model Aircraft.Category  \
0                    Adams            A55           Balloon   
1           Aero Commander            100          Airplane   
2           Aero Commander            112          Airplane   
3           Aero Commander          112TC          Airplane   
4           Aero Commander           500S          Airplane   
...                    ...            ...               ...   
1221             Weatherly           201C          Airplane   
1222                  Wing            D-1          Airplane   
1223         Wootton, John    Aero Canard          Airplane   
1224  Wrb Associates, Llc.  Zodiac 601 XL          Airplane   
1225              Yakovlev         Yak 52          Airplane   

      Total.Fatal.Injuries  Total.Serious.Injuries  
0                      0.0                     0.0  
1                      0.0                     3.0  
2                      0.0                     0.0  
3                      0.0                     0.0  
4                      0.0                     0.0  
...                    ...                     ...  
1221                   0.0                     0.0  
1222                   2.0                     0.0  
1223                   1.0                     0.0  
1224                   1.0                     0.0  
1225                   2.0                     0.0  

[1226 rows x 5 columns]

In [101]:
#combine the damage data and the injury data by merging the sets of data
combined_avaiation_summary = pd.merge(damage_summary, injury_summary, on=["Make", "Model", "Aircraft.Category"], how="inner")

In [102]:
combined_avaiation_summary

Make          Model Aircraft.Category Aircraft.damage  \
0                    Adams            A55           Balloon     Substantial   
1           Aero Commander            100          Airplane       Destroyed   
2           Aero Commander            100          Airplane     Substantial   
3           Aero Commander            112          Airplane     Substantial   
4           Aero Commander          112TC          Airplane     Substantial   
...                    ...            ...               ...             ...   
1562             Weatherly           201C          Airplane     Substantial   
1563                  Wing            D-1          Airplane       Destroyed   
1564         Wootton, John    Aero Canard          Airplane     Substantial   
1565  Wrb Associates, Llc.  Zodiac 601 XL          Airplane       Destroyed   
1566              Yakovlev         Yak 52          Airplane     Substantial   

      Incident.Count  Total.Fatal.Injuries  Total.Serious.Injuries  
0                  1                   0.0                     0.0  
1                  1                   0.0                     3.0  
2                  3                   0.0                     3.0  
3                  1                   0.0                     0.0  
4                  1                   0.0                     0.0  
...              ...                   ...                     ...  
1562               4                   0.0                     0.0  
1563               1                   2.0                     0.0  
1564               1                   1.0                     0.0  
1565               1                   1.0                     0.0  
1566               1                   2.0                     0.0  

[1567 rows x 7 columns]

## Data Visualization

In this section, we shall now make graphs to visualize the analysis we have done

In [ ]:
#filter for safe aircraft models
safe_models = combined_avaiation_summary[
    (combined_avaiation_summary["Total."]]